Eatery/BRBConnectionHandler.swift

```diff
@@ -0,0 +1,169 @@
+        let loginURL = NSURL(string: loginURLString)!
+        loadRequest(NSURLRequest(URL: loginURL))
+    }
+    
+    func failedToLogin() -> Bool {
+        return loginCount > 2
+    }
+    
+    /**
+     
+     - Loads Dining URL Page
+     
+     - TODO: Dining history is not currently being handled.
+     
+     */
+    func getDiningHistory() {
+        if stage != .LoginScreen || stage != .LoginFailed || stage != .Transition {
+            let historyURL = NSURL(string: diningHistoryURLString)!
+            loadRequest(NSURLRequest(URL: historyURL))
+        }
+    }
+    
+    /**
+     
+     - Fetches the HTML for the currently displayed web page and instantiates a new AccountBalance object
+       using the account information on the page.
+     
+     - Does not guarantee that the javascript has finished executing before trying to get account info.
+     
+     */
+    func getAccountBalance() {
+        getHTML { (html: NSString) -> () in
+            self.accountBalance = AccountBalance()
+            let brbHTMLRegex = "<td class=\\\"first-child account_name\\\">BRB Big Red Bucks .*<\\/td><td class=\\\"last-child balance\">\\$[0-9]+.[0-9][0-9]<\\/td>"
+            let swipesHTMLRegex = "<td class=\\\"first-child account_name\\\">A0.*<\\/td><td class=\\\"last-child balance\">[1-9]*[0-9]<\\/td>"
+            let moneyRegex = "[0-9]+(\\.)*[0-9][0-9]"
+            let swipesRegex = ">[1-9]*[0-9]<"
+            if self.stage == .FundsHome {
+                let brbString = self.getFirstRegexMatchFromString(brbHTMLRegex, str: html)
+                let brbs = self.getFirstRegexMatchFromString(moneyRegex, str: brbString)
+                let swipesString = self.getFirstRegexMatchFromString(swipesHTMLRegex, str: html)
+                let swipes = self.getFirstRegexMatchFromString(swipesRegex, str: swipesString)
+                if brbs == "" {
+                    self.getAccountBalance()
+                    return
+                }
+                self.accountBalance.brbs = brbs != "" ? brbs : "0.00"
+                self.accountBalance.swipes = swipes != "" ? swipes[swipes.startIndex.successor()..<swipes.endIndex.predecessor()] : "0"
+            }
+        }
+    }
+    
+    /**
+     
+     - Given a regex string and and a string to match on, returns the first instance of the regex
+       string or an empty string if regex cannot be matched.
+     
+     */
+    func getFirstRegexMatchFromString(regexString: NSString, str: NSString) -> String {
+        do {
+            let regex = try NSRegularExpression(pattern: regexString as String, options: .UseUnicodeWordBoundaries)
+            if let match = regex.firstMatchInString(str as String, options: NSMatchingOptions.WithTransparentBounds , range: NSMakeRange(0, str.length)) {
+                return str.substringWithRange(match.rangeAtIndex(0)) as String
+            }
+        } catch {
+            return ""
+        }
+        return ""
+    }
```

Comment:
Let's convert this to the following since you're not handling any exceptions:
```
let regex = try? NSRegularExpression(pattern: regexString as String, options: .UseUnicodeWordBoundaries)
if let match = regex?.firstMatchInString(str as String, options: NSMatchingOptions.WithTransparentBounds , range: NSMakeRange(0, str.length)) {
    return str.substringWithRange(match.rangeAtIndex(0)) as String
}
return ""
```